internal/gitserver/search/search.go

```
@@ -140,7 +140,7 @@ func (cs *CommitSearcher) feedBatches(ctx context.Context, jobs chan job, result
 	defer func() {
 		// Always call cmd.Wait to avoid leaving zombie processes around.
 		if e := cmd.Wait(); e != nil {
-			err = multierror.Append(err, tryInterpretErrorWithStderr(err, stderrBuf.String()))
+			err = multierror.Append(err, tryInterpretErrorWithStderr(ctx, err, stderrBuf.String()))
```

Won't cmd.Wait also return a context error? Beyond this fix, should we just not return an error to the caller when it is `== context.DeadlineExceeded`

---

Surprisingly, it does not. 

```go
package main

import (
	"context"
	"os/exec"
	"time"
)

func main() {
	ctx, cancel := context.WithCancel(context.Background())
	time.AfterFunc(time.Second, cancel)

	cmd := exec.CommandContext(ctx, "sleep", "1000")
	err := cmd.Start()
	if err != nil {
		panic(err)
	}

	err = cmd.Wait()
	if err != nil {
		panic(err)
	}
}
```

Output:
```
❯ go run .
panic: signal: killed

goroutine 1 [running]:
main.main()
        /Users/ccheek/src/scratch/main.go:21 +0xff
exit status 2
```

---

> Beyond this fix, should we just not return an error to the caller when it is == context.DeadlineExceeded

Do you mean only return nil if `ctx.Err() == context.DeadlineExceeded`?

---

Yes, that's what I mean. If it is a context error, return nil to the caller. I assume that's what we want, right? i.e. timeouts and cancellations are just a stopping signal, not an error.

---

I think that's what I'm already doing, yes? `tryInterpretErrorWithStderr` now checks to see if the context has been canceled, and swallows any errors if it has.

---

Bot for instance on line 161 we're returning `ctx.Err()` to the caller. It'd be good to verify the whole call chain doesn't return these sort of errors (assuming that's intended behaviour).

---

Aah, I see now. Thanks for dumbing it down for me 😄